extensions/resource-deployment/package.json

```diff
@@ -0,0 +1,64 @@
+{
+  "name": "resource-deployment",
+  "displayName": "Azure Data Resource Deployment",
+  "description": "Azure Data Resource Deployment",
+  "version": "0.0.1",
+  "publisher": "Microsoft",
+  "preview": true,
+  "license": "https://raw.githubusercontent.com/Microsoft/azuredatastudio/master/LICENSE.txt",
+  "icon": "images/sqlserver.png",
+  "aiKey": "AIF-5574968e-856d-40d2-af67-c89a14e76412",
+  "engines": {
+    "vscode": "*",
+    "azdata": "^1.6.0"
+  },
+  "activationEvents": [
+    "onCommand:azdata.resource.sql-image.deploy",
+    "onCommand:azdata.resource.sql-bdc.deploy"
+  ],
+  "main": "./out/main",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Microsoft/azuredatastudio.git"
+  },
+  "extensionDependencies": [
+    "Microsoft.mssql"
+  ],
+  "contributes": {
+    "commands": [
+      {
+        "command": "azdata.resource.sql-image.deploy",
+        "title": "%deploy-sql-image-command-name%",
+        "category": "%deploy-resource-command-category%"
+      },
+      {
+        "command": "azdata.resource.sql-bdc.deploy",
+        "title": "%deploy-sql-bdc-command-name%",
+        "category": "%deploy-resource-command-category%"
+      }
+    ],
+    "dataExplorer.actions": [
```

Comment:
Should we be using menu contribution instead here? I'm wondering how this will scale out for other entry points. Normally we'd just add it as a "Menu" contribution, and other than `isPrimary` the definition seems to match that well.

If we are doing something special here (e.g. adding a new pattern of primary vs. secondary actions) we might want to make this generic, either by adding support in the existing contribution points or making this more generic than just the data explorer. Let's think of how this would impact, for example:
- Notebook toolbar contribution, where we want to have primary actions on left, dropdowns in middle, and 2ndary actions (some as icons, some in "...") on the right
- notebook cell "on hover" menu that will require having 2-3 primary actions and a "..." for more. This would replace current "..." menu.

I would propose just extending menu contribution with our own syntax for extra properties. That's the lowest cost and keeps us aligned overall, but happy to discuss.

Reply:
just changed my implementation. if in the future we need to surface as primary actions, we just need to change the group name.